src/cclib/io/wfxwriter.py

```diff
@@ -0,0 +1,488 @@
+
+    def _keywords(self):
+        """Return one of GTO, GIAO, CSGT keyword."""
+        # AIMALL currently supports on GTO keyword.
+        return self.keyword
+
+    def _no_of_nuclei(self):
+        """Number of nuclei in the molecule."""
+        return len(self.ccdata.atomnos)
+
+    def _no_of_prims(self):
+        """Number of Primitives."""
+        nprims = 0
+        for atom in self.ccdata.gbasis:
+            for prims in atom:
+                nprims += num_orb[prims[0]] * len(prims[1])
+        return nprims
+
+    def _no_of_mos(self):
+        """Number of occupied MOs."""
+        return int(max(self.ccdata.homos)) + 1
+
+    def _no_of_perturbations(self):
+        """Number of Perturbation.
+
+        This is usually zero.  For GIAO it should be 3
+        (corresponding to Lx, Ly and Lz), and
+        for CSGT it should be 6
+        (corresponding to Lx, Ly, Lz, Px, Py and Pz).
+        """
+        if 'GIAO' in self._keywords():
+            return 3
+        elif 'CSGT' in self._keywords():
+            return 6
+        return 0
+
+    def _nuclear_names(self):
+        """Names of nuclei present in the molecule.
+
+        O1
+        H2
+        H3
+        """
+        return [self.pt.element[Z]+str(i) for i, Z in
+                enumerate(self.ccdata.atomnos, start=1)]
+
+    def _atomic_nos(self):
+        """Atomic numbers of elements."""
+        return [str(Z) for Z in self.ccdata.atomnos]
+
+    def _nuclear_charges(self):
+        """Nuclear charges."""
+        if hasattr(self.ccdata, 'coreelectrons'):
+            return ['{:22.11E}'.format(Z) for Z in self.ccdata.atomnos - self.ccdata.coreelectrons]
+        else:
+            return ['{:22.11E}'.format(Z) for Z in self.ccdata.atomnos]            
+
+    def _nuclear_coords(self):
+        """Nuclear coordinates in Bohr."""
+        coord_template = '%22.11E %.12E %.12E'
+        to_bohr = lambda x: utils.convertor(x, 'Angstrom', 'bohr')
+        nuc_coords = [coord_template % tuple(to_bohr(coord))
+                      for coord in self.ccdata.atomcoords[-1]]
+        return nuc_coords
+
+    def _net_charge(self):
+        """Net charge on molecule."""
+        return '%22.11E' % self.ccdata.charge
+
+    def _no_electrons(self):
+        """Number of electrons in molecule."""
+        homos = self.ccdata.homos
+        charge = self.ccdata.charge if hasattr(self.ccdata, 'charge') else 0.0
+        if len(homos) > 1:
+            return int(sum(homos)) + 2
+        return int((homos[0] + 1)*2 - (charge % 2))
+
+    def _no_alpha_electrons(self):
+        """Number of alpha electrons in molecule."""
+        return int(numpy.ceil(self._no_electrons() / 2.0))
+
+    def _no_beta_electrons(self):
+        """Number of beta electrons in molecule."""
+        return self._no_electrons() - self._no_alpha_electrons()
+
+    def _spin_mult(self):
+        """Electronic Spin Multiplicity"""
+        return self.ccdata.mult
+
+    def _prim_centers(self):
```

Comment:
Can you get this in an easier way from `atombasis`?

Reply:
`atombasis`. does not directly translate to what is expected to be written in the section.
For instance for water_mp2:
`atombasis` is 
```python
[[0, 1, 2, 3, 4], [5], [6]]
```
and the expected output is
```xml
<Primitive Centers>
1 1 1 1 1 1 1 1 1 1 
1 1 1 1 1 2 2 2 3 3 
3 
</Primitive Centers>
```
The conversion would require knowing the name of orbitals, and hence using `ccData.gbasis`.